### PR TITLE
Feat: Create Test Session Service

### PR DESCRIPTION
**Pull request by carissa-tang**
## Notion ticket link

[Create test session](https://www.notion.so/uwblueprintexecs/Create-test-session-71e3938ba11c473abf3dfdbb5c704702)

## Implementation description
* Wrote `createTestSession` function
* Updated `results` to be an optional attribute
* Fixed types of ids in test and testSession models
* Added missing attributes for testSession model
* Wrote one successful test

- *Not sure what, if any, failing case needs to be covered

## Steps to test
1.

## What should reviewers focus on?
* 

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/testUtils/testSession.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const mockTestSession = {
+    test: "1",
+    teacher: "2",
+    school: "3",
+    grade_level: 4,
+    results: [],
```
Comment (carissa-tang) [inline backend/testUtils/testSession.ts:6]: When we create a test session, would we supply an empty array for `results`? And if that's the case, what would the type for `results` be? Cause right now as `results: [Result]`, it is requiring at least one Result object @joyce-shi

**Comment by carissa-tang** [comment]
Side question... should we be using camelCase?

File: backend/testUtils/testSession.ts (hunk 1 above)
Comment (joyce-shi) [reply backend/testUtils/testSession.ts]: yes! we should start the array as empty and then populate it as student results are received. given this, do you mind updating the `Result` schema so the the field is optional? (I think just set `required: false`)

File: backend/services/interfaces/testSessionService.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+export interface CreateTestSessionRequestDTO {
```
Comment (mahadzaryab1) [inline backend/services/interfaces/testSessionService.ts]: nit: can we document these interfaces?

File: backend/services/interfaces/testSessionService.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  test: string;
+  teacher: string;
+  school: string;
+  grade_level: number;
+  access_code: string;
+  start_time: Date;
+}
+
+export interface CreateTestSessionResponseDTO {
```
Comment (mahadzaryab1) [inline backend/services/interfaces/testSessionService.ts]: same

File: backend/models/test.model.ts [hunk 4]
```diff
@@ -87,7 +87,8 @@ const TestSchema: Schema = new Schema(
       required: true,
     },
     admin: {
-      type: { type: mongoose.Schema.Types.ObjectId, ref: "User" },
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "User",
```
Comment (joyce-shi) [inline backend/models/test.model.ts:91]: thanks for catching this syntax error!

File: backend/models/testSession.model.ts [hunk 5]
```diff
@@ -58,23 +58,28 @@ export interface TestSession extends Document {
...
     teacher: {
-      type: { type: mongoose.Schema.Types.ObjectId, ref: "User" },
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "User",
       required: true,
     },
     school: {
-      type: { type: mongoose.Schema.Types.ObjectId, ref: "School" },
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "School",
```
Comment (joyce-shi) [inline backend/models/testSession.model.ts:82]: 🥺 ^^ thanks again LOL

**Review by mahadzaryab1 — COMMENTED** [review]
couple of small comments

**Review by joyce-shi — APPROVED** [review]
LGTM, other than what we discussed async. +1 to Mahad's documentation comments, but other than that good to 🛳 !

**Pull request merged**